src/chips/sam4l/flashcalw.rs

```diff
@@ -0,0 +1,1103 @@
+        }
+            self.error_status.replace(error_status);
+    }
+
+    pub fn set_gp_fuse_bit(&self, gp_fuse_bit : u32, value : bool) {
+        if value {
+            self.erase_gp_fuse_bit(gp_fuse_bit, false);    
+        } else {
+            self.write_gp_fuse_bit(gp_fuse_bit, false);    
+        }
+    }
+
+    pub fn set_gp_fuse_bitfield(&self, mut pos: u32, mut width : u32, mut value : u64) {
+        let mut error_status : u32 = 0;
+
+        pos &= 0x3F;
+        width = min(width, 64);
+
+        for gp_fuse_bit in pos..(pos + width) {
+            self.set_gp_fuse_bit(gp_fuse_bit, value & 0x01 != 0);
+            error_status |= self.error_status.take().unwrap();
+            value >>= 1;
+        }
+        self.error_status.replace(error_status);
+    }
+
+    pub fn set_gp_fuse_byte(&self, gp_fuse_byte : u32, value : u8) {
+        let mut error_status : u32;
+        match value {
+            0xFF => {
+                self.erase_gp_fuse_byte(gp_fuse_byte, false);    
+            },
+            0x00 => {
+                self.write_gp_fuse_byte(gp_fuse_byte, 0x00);
+            },
+            _ => {
+                self.erase_gp_fuse_byte(gp_fuse_byte, false);
+                error_status = self.error_status.take().unwrap();
+                self.write_gp_fuse_byte(gp_fuse_byte, value);
+                error_status |= self.error_status.take().unwrap();
+                self.error_status.replace(error_status);
+            }
+        };
+
+    }
+
+    pub fn set_all_gp_fuses(&self, value : u64) {
+        let mut error_status : u32;
+
+        match value {
+            GP_ALL_FUSES_ONE => {
+                self.erase_all_gp_fuses(false);  
+            },
+            0u64 => {
+                self.write_all_gp_fuses(0u64);  
+            },
+            _ => {
+                self.erase_all_gp_fuses(false);
+                error_status = self.error_status.take().unwrap();
+                self.write_all_gp_fuses(value);
+                error_status |= self.error_status.take().unwrap();
+                self.error_status.replace(error_status);
+            }
+        }
+    }
+    
+    ///Flashcalw Access to Flash Pages
+    pub fn clear_page_buffer(&self) {
+        self.issue_command(FlashCMD::CPB, -1);    
+    }
+
+    pub fn is_page_erased(&self) -> bool {
+        let registers : &mut Registers = unsafe {
+            mem::transmute(self.registers)     
+        };
+        let status = registers.status.get();
+
+        (status & get_bit!(5)) != 0
+    }
+
+    pub fn quick_page_read(&self, page_number : i32) -> bool {
+        self.issue_command(FlashCMD::QPR, page_number);
+        self.is_page_erased()
+    }
+
+    pub fn flashcalw_erase_page(&self, page_number : i32, check : bool) -> bool {
+        let mut page_erased = true;
+
+        self.issue_command(FlashCMD::EP, page_number);
+        if check {
+            let mut error_status : u32 = self.error_status.take().unwrap();
```

Comment:
Yeah, `self.error_status` definitely needs to be a `Cell`, you're emulating a `Cell` everywhere you use it.